rs/execution_environment/src/history.rs

```diff
@@ -159,8 +181,60 @@ impl IngressHistoryWriter for IngressHistoryWriterImpl {
                 status
             );
         }
+
         use IngressState::*;
         use IngressStatus::*;
+
+        // Computes the duration elapsed since the time when a message was included into
+        // a block (approximated by block time); and since its induction.
+        let message_latencies_seconds = |message_id| {
+            let received_time = self.received_time.read().unwrap();
+            received_time.get(message_id).map(|timer| {
+                (
+                    system_time_now()
```

Comment:
Using system time here will mean things are non-deterministic across replicas? I suppose for metrics this is not a problem? 

Maybe we can somewhere add a comment with a warning to make sure nobody uses these things in a different context.